interactions/client/client.py

```diff
@@ -1807,7 +1807,12 @@ async def _dispatch_interaction(self, event: RawGatewayEvent) -> None:  # noqa:
 
                 if auto_opt := getattr(ctx, "focussed_option", None):
                     if autocomplete := ctx.command.autocomplete_callbacks.get(str(auto_opt.name)):
-                        callback = autocomplete
+                        if ctx.command.has_binding:
+                            callback = CallbackObject()
+                            callback.callback = autocomplete
+                            callback._binding = ctx.command._binding
+                        else:
+                            callback = autocomplete
```

Comment:
This is more overhead than it needs to. Instead of making a new callback object, using `ctx.command.call_with_binding` (with `functools.partial`) is likely much better.

Reply:
Great. I'll try to figure out that syntax and resubmit.